inc/commoninjectionlib.class.php

```diff
@@ -955,7 +955,7 @@ private function setValueForItemtype($itemtype, $field, $value, $fromdb = false)
             }
         } else { // First value
             if (empty($value)) {
-                if (isForeignKeyField($field) || (strpos($field, 'is_') !== false)) {
+                if (isForeignKeyField($field) || (strpos($field, 'is_') !== false) || (method_exists($injectionClass, 'isNullable') && !$injectionClass->isNullable($field))) {
```

Comment:
Rather than testing if the function exists, it should be added to the `PluginDataInjectionInjectionInterface`. This way, it can benefit all other injection classes that might need it at some point in the future.

Reply:
UP @Lainow